api/types/saml_idp_service_provider_test.go

```diff
@@ -230,6 +231,31 @@ func TestNewSAMLIdPServiceProvider(t *testing.T) {
 			preset:             "",
 			expectedRelayState: "",
 		},
+		{
+			name:       "http launch url",
+			entityID:   "IAMShowcase",
+			acsURL:     "https:/test.com/acs",
+			launchURLs: []string{"http://test.com/myapp"},
+			errAssertion: func(t require.TestingT, err error, i ...interface{}) {
+				require.ErrorContains(t, err, "invalid scheme")
+			},
+		},
+		{
+			name:       "empty launch URLs",
+			entityID:   "IAMShowcase",
+			acsURL:     "https:/test.com/acs",
+			launchURLs: []string{""},
+			errAssertion: func(t require.TestingT, err error, i ...interface{}) {
+				require.ErrorContains(t, err, "invalid scheme")
+			},
+		},
+		{
+			name:         "valid launch URLs",
+			entityID:     "IAMShowcase",
+			acsURL:       "https:/test.com/acs",
+			launchURLs:   []string{"https://test.com/myapp", "https://anysubdomain.test.com/myapp"},
+			errAssertion: require.NoError,
```

Comment:
Perhaps we should validate `acsURL` too? Here it is invalid due to missing `/`: `https:/` instead of `https://`.

Reply:
Good eyes :)

I fixed that typo and used a const value in https://github.com/gravitational/teleport/pull/46619/commits/d72a38caa7302b7edbdd180f7c435223ce86c0eb

We do validation for the `acsURL` field but that happens here in the `/lib/services` and only enforced on create/update actions. https://github.com/gravitational/teleport/blob/4ec095620ded2552e49819dfd6f8c1c3d9f4ecb9/lib/services/saml_idp_service_provider.go#L168-L175

That was introduce couple of releases after the `v12` (where SAML IdP was introduced). So the validation was added there instead of checkAndSetDefaults to avoid bricking cluster that may already have acs endpoint with `http` scheme.